lib/beaker/dsl/helpers/web_helpers.rb

```diff
@@ -52,9 +53,17 @@ def fetch_http_file(base_url, file_name, dst_dir)
           else
             logger.notify "Fetching: #{src}"
             logger.notify "  and saving to #{dst}"
-            open(src, :allow_redirections => :all) do |remote|
-              File.open(dst, "w") do |file|
-                FileUtils.copy_stream(remote, file)
+            begin
+              open(src, :allow_redirections => :all) do |remote|
+                File.open(dst, "w") do |file|
+                  FileUtils.copy_stream(remote, file)
+                end
+              end
+            rescue OpenURI::HTTPError => e
+              if e.message =~ /404.*/
```

Comment:
Yeah, I think I saw some rubydocs for this class that only had those methods, but then I saw another that had a bunch, and I couldn't tell which we had here.  This is totally fine, I was just thinking it was better to use the exit code field directly since that what you're after (not the message).